app/models/alchemy/page.rb

```diff
@@ -350,41 +333,66 @@ def publish!
     # @param [TreeNode]
     #   A tree node with new lft, rgt, depth, url, parent_id and restricted indexes to be updated
     #
-    def update_node!(node)
-      hash = {lft: node.left, rgt: node.right, parent_id: node.parent, depth: node.depth, restricted: node.restricted}
+    # def update_node!(node)
+    #   hash = {lft: node.left, rgt: node.right, parent_id: node.parent, depth: node.depth, restricted: node.restricted}
+    #
+    #   if !self.redirects_to_external? && self.urlname != node.url
+    #     LegacyPageUrl.create(page_id: self.id, urlname: self.urlname)
+    #     hash.merge!(urlname: node.url)
+    #   end
+    #
+    #   update_columns(hash)
+    # end
 
-      if Config.get(:url_nesting) && !self.redirects_to_external? && self.urlname != node.url
-        LegacyPageUrl.create(page_id: self.id, urlname: self.urlname)
-        hash.merge!(urlname: node.url)
-      end
+    # Returns the url for menu node
+    def alchemy_node_url
+      read_attribute(:urlname)
+    end
+
+    # The url of the pages's parent
+    def parent_urlname
+      parent.try(:urlname)
+    end
 
-      update_columns(hash)
+    # Return the first node this page is attached at or nil
+    def node
+      nodes.first
+    end
+
+    # If the page has a parent it returns the node of it
+    def parent_node
+      parent.try(:node)
+    end
+
+    # Recursily get all parents
+    # TODO: Implement this as a single SQL query
+    # Maybe use a parents_ids array column as cache to select from
+    def parents
+      @parents ||= begin
+        arr = []
+        par = self.parent
+        while par do
```

Comment:
Style/WhileUntilDo: Do not use `do` with multi-line `while`. (https://github.com/bbatsov/ruby-style-guide#no-multiline-while-do)